packages/greedy_infomax/src/nupic/research/frameworks/greedy_infomax/models/classification_model.py

```diff
@@ -48,12 +48,47 @@ def __init__(self, in_channels=256, num_classes=10):
             "layer1", nn.Linear(self.in_channels, num_classes, bias=True)
         )
 
+    def forward(self, x):
+        # detach x just in case it's still connected to active parts
+        # of the computation graph
+        batch_size = x.shape[0]
+        x = x.detach()
+        x = self.avg_pool(x).squeeze()
+        x = self.model(x).squeeze()
+        return x.view(batch_size, -1)
+
+
+class FlattenClassifier(nn.Module):
+    """
+    A simple multilayer perceptron classification head which outputs a distribution
+    over possible class labels. This is used in the supervised phase of Greedy
```

Comment:
It's not clear from the docstring that you're taking _z_'s corresponding to different parts of an image and flattening them into 1 long vector; my initial thought was you're taking the output of a conv layer and removing the _x_ or _y_ dimension. Can you please make this clear?

Reply:
Yes, I'll explain this better